### ISSUE TITLE
OS X: Capturing input on a child process

### ISSUE THREAD

**Issue by kapstok** [TolikPylypchuk/SharpHook#108]
Hello,

I have created a MAUI app for Mac catalyst and used SharpHook to capture keyboard input. When I compile and run the app using the two commands below, capturing input works as expected.

```bash
dotnet build ./MAUI\ DEMO/MAUI\ DEMO.csproj -f net7.0-maccatalyst
./MAUI\ DEMO/bin/Debug/net7.0-maccatalyst/maccatalyst-arm64/MAUI\ DEMO.app/Contents/MacOS/MAUI\ DEMO
```

But building and running the application with different methods do seem to cause issues. Keyboard input is not captured when running the command below. Using Rider to compile and run gives issues as well. Moreover, when I run the application by starting a process in a different MAUI app, the input won't be captured as well.

```bash
dotnet build -t:Run ./MAUI\ DEMO/MAUI\ DEMO.csproj -f net7.0-maccatalyst
```

My assumption is that SharpHook won't capture input if the application in is used in, is a child process. I haven't tested this problem on other OSes, so this problem could as well occur on Linux or Windows.

**Comment by TolikPylypchuk** [comment]
Hi! Thanks for posting this issue!

The first thing that comes to mind is whether you've enabled access to accessibility APIs for the correct applications. As far as I remember, if you run your application from the terminal, then you need to give this access to the terminal itself. Same with Rider - I think you should give this access to Rider.

On the other hand, you'd get a `HookException` with `AxApiDisabled` and a system popup asking you to enabled it if that were the case so you'd most probably see it.

If the accessiblity API is enabled for these applications, then I will look into it as in that case it's a bug.

**Comment by kapstok** [comment]
I tried to re-run the application from Finder several times. Everytime I run the application, the accessibility API permission request pops up, even though I already gave permission. Managed to fix it by [resetting App permissions](https://apple.stackexchange.com/a/406124). It's very inconventient, because I have to reset the App permissions everytime I recompile, but it shows that this is not a bug in SharpHook. I hope I find a more conventient solution to solve this problem instead of having to execute a command after every successful recompilation. Because this could be tedious especially if you publish a new version of your software and the app permissions have to be reset by the end-user.
